base_placement_plugin/src/create_marker.cpp

```diff
@@ -0,0 +1,313 @@
+  for (BasePoseJoint::iterator it = baseJoints.begin(); it !=baseJoints.end(); ++it)//Creating each interactive marker
+  {
+    int i=std::distance(baseJoints.begin(), it);
+    bool no_joint_solution = false; //flag for checking if for this robot model no ik solution exist
+    geometry_msgs::Pose base_pose = it->second;
+    std::vector<double> joint_soln = it->first;
+    if(std::equal(joint_soln.begin()+1, joint_soln.end(), joint_soln.begin()))
+    {
+      no_joint_solution = true;
+    }
+
+    moveit::core::RobotStatePtr robot_state_(new moveit::core::RobotState(robot_model_));//Calling robot_state in each iteration
+    for(int k=0;k<joint_soln.size();k++)
+    {
+      robot_state_->setJointPositions(joint_names[k], &(joint_soln[k])); //Setting joint values to arm joints
+    }
+    robot_state_->update();
+    std::vector<std::string> full_link_names = robot_model_->getLinkModelNames(); //Collection of all the links for arm and gripper
+    //ROS_INFO("Number of links: %lu ",full_link_names.size());
+    visualization_msgs::MarkerArray full_link_markers;
+    robot_state_->getRobotMarkers( full_link_markers, full_link_names);//Obtain the current markers
+    //ROS_INFO("Number of markers: %lu",full_link_markers.markers.size());
+
+    const std::string& ee_parent_link_name =  full_link_names[0];//root link of the arm
+    Eigen::Affine3d tf_root_to_first_link = robot_state_->getGlobalLinkTransform(ee_parent_link_name);//transform of the root link
+    Eigen::Affine3d tf_first_link_to_root = tf_root_to_first_link.inverse();//inverse of the transform
+    Eigen::Affine3d base_tf;
+    tf::poseMsgToEigen(base_pose, base_tf);//base pose in tf
+
+    visualization_msgs::InteractiveMarker iMarker; //new interactive marker
+    iMarker.header.frame_id = "base_link";
+    iMarker.pose = base_pose;//the pose of the imarker is the pose of the base
+    iMarker.scale = 0.2;
+    std::string name = "robot_model";
+    std::string description = "robot model";
+    iMarker.name = name+boost::lexical_cast<std::string>(i);
+    iMarker.description = description+boost::lexical_cast<std::string>(i);//each imarker has different name and description
+    visualization_msgs::InteractiveMarkerControl robotModelControl;
+    //Setting color for individual robot
+    double r = unifRand();
+    double g = unifRand();
+    double b = unifRand();
+    for(std::size_t j=0; j<full_link_markers.markers.size();j++)
+    {
+      full_link_markers.markers[j].header.frame_id = "base_link";
+      full_link_markers.markers[j].type = visualization_msgs::Marker::MESH_RESOURCE;
+      full_link_markers.markers[j].mesh_use_embedded_materials = true;
+      full_link_markers.markers[j].id = i*(j+5);
+      full_link_markers.markers[j].header.stamp = ros::Time::now();
+      full_link_markers.markers[j].ns = "robot_links";
+      full_link_markers.markers[j].lifetime = ros::Duration(40.0);
```

Comment:
@akgoins  I have created a new function. So now the create_marker is pretty modular. Please review the PR with the new changes.